benchmarks/nx-cugraph/pytest-based/create_results_summary_page.py

```diff
@@ -0,0 +1,245 @@
+
+    # Organize all benchmark runs by the following hierarchy: algo -> backend -> dataset
+    benchmarks = {}
+
+    # Populate benchmarks dir from .json files
+    for json_file in logs_dir.glob("*.json"):
+        # print(f"READING {json_file}")
+        try:
+            data = json.loads(open(json_file).read())
+        except json.decoder.JSONDecodeError:
+            # print(f"PROBLEM READING {json_file}, skipping.")
+            continue
+
+        for benchmark_run in data["benchmarks"]:
+            # example name: "bench_triangles[ds=netscience-backend=cugraph-preconverted]"
+            name = benchmark_run["name"]
+
+            algo_name = name.split("[")[0]
+            if algo_name.startswith("bench_"):
+                algo_name = algo_name[6:]
+            # special case for betweenness_centrality
+            match = k_patt.match(name)
+            if match is not None:
+                algo_name += f", k={match.group(1)}"
+
+            match = dataset_patt.match(name)
+            if match is None:
+                raise RuntimeError(
+                    f"benchmark name {name} in file {json_file} has an unexpected format"
+                )
+            dataset = match.group(1)
+            if dataset.endswith("-backend"):
+                dataset = dataset[:-8]
+
+            match = backend_patt.match(name)
+            if match is None:
+                raise RuntimeError(
+                    f"benchmark name {name} in file {json_file} has an unexpected format"
+                )
+            backend = match.group(1)
+            if backend == "None":
+                backend = "networkx"
+
+            runtime = benchmark_run["stats"]["mean"]
+            benchmarks.setdefault(algo_name, {}).setdefault(backend, {})[
+                dataset
+            ] = runtime
+
+    # dump HTML table
+    ordered_datasets = [
+        "netscience",
+        "email_Eu_core",
+        "cit-patents",
+        "hollywood",
+        "soc-livejournal1",
+    ]
+
+    print(
+        """
+    <html>
+    <head>
+    <style>
+        table {
+            table-layout: fixed;
+            width: 100%;
+            border-collapse: collapse;
+        }
+        tbody tr:nth-child(odd) {
+            background-color: #ffffff;
+        }
+        tbody tr:nth-child(even) {
+            background-color: #d3d3d3;
+        }
+        tbody td {
+            text-align: center;
+            color: black;
+        }
+        th,
+        td {
+            padding: 10px;
+        }
+        .footer {
+            background-color: #f1f1f1;
+            padding: 10px;
+            font-size: 12px;
+            color: black;
+            width: 100%;
+        }
+        .text-indent {
+            text-indent: 20px; /* Indents the first line of the text by 30px */
+        }
+    </style>
+    </head>
+    <table>
+    <thead>
+    <tr>
+        <th></th>"""
+    )
+    for ds in ordered_datasets:
+        print(f"      <th>{ds}</th>")
```

Comment:
Can you make this a multi-line cell and also include the number of nodes and edges and if it's directed or undirected?